### PR TITLE
[go_router] Adds missed popping log

### PR DESCRIPTION
**Pull request by ycherniavskyi**
It is confusing that when using push/pop functionality of `go_router`, especially when pushing cascading, in log you only see `pushing ...` messages but no `poping ...` ones. So this tiny PR aims to change it.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by stuartmorgan** [comment]
Thanks for the submission! You haven't completed some required steps, so I'm going to mark this as a draft for now; please mark as ready for review once those are addressed:

> * [ ]  I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy](https://dart.dev/tools/pub/versioning), or this PR is [exempt from version changes](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates).
> * [ ]  I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style).
> [...]
> * [ ]  I added new tests to check the change I am making, or this PR is [test-exempt](https://github.com/flutter/flutter/wiki/Tree-hygiene#tests).

**Comment by ycherniavskyi** [comment]
@stuartmorgan I have no clue how I can write test for log output and currently there no any [test-exempt](https://github.com/flutter/flutter/wiki/Tree-hygiene#tests) like logging changes/adds.
Could you please suggest me what to do?

File: packages/go_router/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 3.1.1
+
+- Adds missed poping log.
```
Comment (chunhtai) [inline packages/go_router/CHANGELOG.md:3]: nit: poping -> popping, here and everywhere in this pr

Comment (ycherniavskyi) [reply packages/go_router/CHANGELOG.md]: Ups. Fixed by rebase.

**Comment by stuartmorgan** [comment]
@chunhtai @johnpryan It looks like this is ready for re-review.

File: packages/go_router/lib/src/go_router_delegate.dart [hunk 2]
```diff
@@ -174,6 +174,7 @@ class GoRouterDelegate extends RouterDelegate<Uri>
 
   /// Pop the top page off the GoRouter's page stack.
   void pop() {
+    log.info('popping $location');
```
Comment (johnpryan) [inline packages/go_router/lib/src/go_router_delegate.dart:177]: This should be in `go_router.dart`, where the other log statements are.

**Review by johnpryan — CHANGES_REQUESTED** [review]

File: packages/go_router/lib/src/go_router_delegate.dart (hunk 2 above)
Comment (ycherniavskyi) [reply packages/go_router/lib/src/go_router_delegate.dart:177]: For sure log statements was in `go_router_delegate.dart` when this PR was created 😉. Ok I will rebase it.

**Review by johnpryan — APPROVED** [review]

**Pull request merged**